### PR TITLE
Add custom IRecipient deserializer and unit tests for Email context

### PR DESCRIPTION
**Pull request by bingenito**

### CODE REVIEW

File: src/Fdc3.Json/Serialization/IRecipientJsonConverter.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+
+using MorganStanley.Fdc3.Context;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+
+namespace MorganStanley.Fdc3.Json.Serialization
+{
+    public  class IRecipientJsonConverter : JsonConverter
```
Comment (psmulovics) [inline src/Fdc3.Json/Serialization/IRecipientJsonConverter.cs:22]: I get that the converter is for IRecipient, but now this looks like an interface itself

**Review by psmulovics — APPROVED** [review]

File: src/Fdc3.Json/Serialization/IRecipientJsonConverter.cs (hunk 1 above)
Comment (bingenito) [reply src/Fdc3.Json/Serialization/IRecipientJsonConverter.cs:22]: Think I should drop the I?

Comment (psmulovics) [reply src/Fdc3.Json/Serialization/IRecipientJsonConverter.cs:22]: Add an underscore? 😂😂

Comment (bingenito) [reply src/Fdc3.Json/Serialization/IRecipientJsonConverter.cs]: Renamed to RecipientJsonConverter.

**Review by psmulovics — APPROVED** [review]

**Pull request merged**